app/components/header.tsx

```diff
@@ -1,25 +1,47 @@
+"use client";
+
+import { socket } from "@/socket";
 import {
   SignedIn,
   SignedOut,
   SignInButton,
   SignOutButton,
 } from "@clerk/nextjs";
+import { UsersIcon } from "lucide-react";
+import { useEffect, useState } from "react";
 
 export const HEADER_HEIGHT = 64;
 
 export function Header() {
+  const [clientsCount, setClientsCount] = useState(0);
+
+  useEffect(() => {
+    socket.on("clientsCount", setClientsCount);
+    return () => {
+      socket.off("clientsCount", setClientsCount);
+    };
+  }, []);
```

Comment:
if we're going to be doing a lot with the socket, like this, maybe we can extract this to a hook?

Reply:
moved to `useEvent` hook